src/current/v25.2/row-level-security.md

```diff
@@ -0,0 +1,561 @@
+
+- Employees can view their own record.
+- Managers can view the records of their direct reports.
+- Members of the `hr_department` role can view all records.
+
+#### Set up fine-grained access control schema
+
+First, define the `hr_department` role and `employees` table, add some data, and grant basic table access:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- Create a role needed for the example policies.
+-- Note: In a real scenario, manage roles appropriately.
+-- This may require admin privileges not available to all users.
+CREATE ROLE hr_department;
+CREATE ROLE manager;
+CREATE ROLE employee;
+
+-- Create roles for employees.
+CREATE ROLE alice;
+CREATE ROLE bob;
+CREATE ROLE carol;
+CREATE ROLE david;
+CREATE ROLE edward;
+
+GRANT hr_department to edward;
+GRANT manager to alice, carol;
+GRANT employee to alice, bob, carol, david, edward;
+
+-- Assume roles 'hr_department' and potentially others exist.
+-- Usernames are assumed to match the 'username' column for simplicity.
+CREATE TABLE employees (
+    id SERIAL PRIMARY KEY,
+    username TEXT UNIQUE NOT NULL,
+    full_name TEXT NOT NULL,
+    manager_username TEXT,
+    salary NUMERIC(10, 2) NOT NULL
+);
+
+-- Sample Data
+INSERT INTO employees (username, full_name, manager_username, salary) VALUES
+('alice', 'Alice Smith', NULL, 120000),
+('bob', 'Bob Jones', 'alice', 80000),
+('carol', 'Carol White', 'alice', 85000),
+('david', 'David Green', 'carol', 70000),
+('edward', 'Edward Scissorhands', 'alice', 70000);
+
+-- Grant basic table access (RLS will refine row access)
+GRANT SELECT, INSERT, UPDATE, DELETE ON employees TO hr_department;
+GRANT SELECT ON employees TO manager;
+GRANT SELECT ON employees TO employee;
+~~~
+
+#### Enable row-level security for fine-grained access control
+
+Next, enable row-level security using the [`ALTER TABLE ... ENABLE ROW LEVEL SECURITY`]({% link {{ page.version.version }}/alter-table.md %}#enable-disable-row-level-security) statement. Optionally, you may want to ensure that the table owner is also subject to RLS using [`ALTER TABLE ... FORCE ROW LEVEL SECURITY`]({% link {{ page.version.version }}/alter-table.md %}#force-unforce-row-level-security).
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER TABLE employees ENABLE ROW LEVEL SECURITY;
+-- Optional: Ensure owner is also subject to policies if needed
+-- ALTER TABLE employees FORCE ROW LEVEL SECURITY;
+~~~
+
+#### Define row-level security policies for fine-grained access control
+
+Next, define RLS policies on the table. Policy 1 allows HR full access to the table:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE POLICY hr_access ON employees
+    FOR ALL -- Applies to SELECT, INSERT, UPDATE, DELETE
+    TO hr_department
+    USING (true) -- No row restriction for HR
+    WITH CHECK (true); -- No check restriction for HR
+~~~
+
+Policy 2 allows employees to view their own record.
```

Comment:
```suggestion
The `self_access` policy allows employees to view their own record.
```

Reply:
done (went with "the following" to avoid restating policy names)